### PR TITLE
Fix linting on v3.3.x

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Backport of #18091, plus correction of flake8 error (see commits).

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [n/a] New features are documented, with examples if plot related
- [n/a] Documentation is sphinx and numpydoc compliant
- [n/a] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [n/a] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 from matplotlib.axes import Axes, SubplotBase, subplot_class_factory
 from matplotlib.blocking_input import BlockingMouseInput, BlockingKeyMouseInput
-from matplotlib.gridspec import GridSpec, SubplotSpec
+from matplotlib.gridspec import GridSpec, SubplotSpec  # noqa: F401
```
Comment (timhoffm) [inline lib/matplotlib/figure.py:31]: I think we favor per-file-ignores in `.flake8` over `# noqa` comments to not clutter the code base.

AFAIK this was discussed at some point but is not formally written down.

Comment (QuLogic) [reply lib/matplotlib/figure.py:31]: I'd rather not, as it's only one import line, not all over the file, and it's already been removed in `master`.

Comment (dopplershift) [reply lib/matplotlib/figure.py:31]: What was removed on master?

Comment (QuLogic) [reply lib/matplotlib/figure.py:31]: This `SubplotSpec` import: https://github.com/matplotlib/matplotlib/pull/18029/files#diff-c3ce78ddcc28c54e04ade080c149c1b1L32

Comment (dopplershift) [reply lib/matplotlib/figure.py:31]: Oh. Then yeah, let's just `noqa` it.

**Comment by timhoffm** [comment]
CI failure Will be fixed by #18104.

**Comment by jkseppan** [comment]
The restarted Travis runs passed, so I think this is safe to merge

**Pull request merged**